src/lib/strapi.ts

```diff
@@ -279,52 +281,45 @@ export class Strapi {
    *
    * @param  {string} contentType - Content type's name pluralized
    * @param  {AxiosRequestConfig["params"]} params? - Filter and order queries
-   * @returns Promise<T>
+   * @returns Promise<StrapiResponse<T>>
    */
   public find<T>(
     contentType: string,
     params?: AxiosRequestConfig["params"]
```

Comment:
We could create a custom type for the params to provide Type specific intellisense based on the types we can generate. I think we could actually incldue the generation in the sdk by call the content-type-builder api to get the content types schemas and generate a file that we can base the params types on afterwards and even the contentType value string

Reply:
Totally agree with that! The problem with the generation in the sdk for request params type is that we don't have them before users run the SDK so we will not be able to build the SDK with those custom type 🤔 

If you know a way to do it that would be really great 😄